VultisigApp/VultisigApp/Services/Blowfish/BlowfishService.swift

```diff
@@ -28,30 +28,96 @@ struct BlowfishService {
         }
         
-        let blowfishRequest = BlowfishRequest(
-            userAccount: userAccount,
-            metadata: BlowfishRequest.BlowfishMetadata(origin: origin),
-            txObjects: txObjects,
-            simulatorConfig: simulatorConfig
-        )
-        
-        let endpoint = Endpoint.fetchBlowfishTransactions(chain: supportedChain, network: supportedNetwork)
-        let headers = ["X-Api-Version" : "2023-06-05"]
-        let body = try JSONEncoder().encode(blowfishRequest)
-        let dataResponse = try await Utils.asyncPostRequest(urlString: endpoint, headers: headers, body: body)
-        let response = try JSONDecoder().decode(BlowfishResponse.self, from: dataResponse)
-        
-        return response
+        do {
+            
+            let blowfishRequest = BlowfishRequest(
+                userAccount: userAccount,
+                metadata: BlowfishRequest.BlowfishMetadata(origin: origin),
+                txObjects: txObjects,
+                simulatorConfig: simulatorConfig
+            )
+            
+            let endpoint = Endpoint.fetchBlowfishTransactions(chain: supportedChain, network: supportedNetwork)
+            let headers = ["X-Api-Version" : "2023-06-05"]
+            let body = try JSONEncoder().encode(blowfishRequest)
+            let dataResponse = try await Utils.asyncPostRequest(urlString: endpoint, headers: headers, body: body)
+            let response = try JSONDecoder().decode(BlowfishResponse.self, from: dataResponse)
+            
+            return response
+            
+        } catch {
+            
+            print(error.localizedDescription)
+            return nil
+            
+        }
     }
     
+    enum DecodingCustomError: Error {
+        case dataCorrupted(DecodingError.Context)
+        case keyNotFound(CodingKey, DecodingError.Context)
+        case valueNotFound(Any.Type, DecodingError.Context)
+        case typeMismatch(Any.Type, DecodingError.Context)
+    }
+    
+    func scanSolanaTransactions
+    (
+        userAccount: String,
+        origin: String,
+        transactions: [String]
+    ) async -> BlowfishResponse? {
+        
+        do {
+            
+            let blowfishRequest = BlowfishSolanaRequest(
+                userAccount: userAccount,
+                metadata: BlowfishSolanaRequest.BlowfishMetadata(origin: origin),
+                transactions: transactions
+            )
+            
+            let endpoint = Endpoint.fetchBlowfishSolanaTransactions()
+            let headers = ["X-Api-Version" : "2023-06-05"]
+            let body = try JSONEncoder().encode(blowfishRequest)
+            let dataResponse = try await Utils.asyncPostRequest(urlString: endpoint, headers: headers, body: body)
+            
+            do {
+                let response = try JSONDecoder().decode(BlowfishResponse.self, from: dataResponse)
+                return response
+            } catch let DecodingError.dataCorrupted(context) {
```

Comment:
still do not understand this json decoding handlers here

Reply:
errorDescription has same text inside

Reply:
That is because Blowfish does not have a pattern or return in their JSON; each transaction returns a different JSON, like a tx for EVM, which is different from an EVM token, which is different from SOLANA, which is different for SOLANA tokens.
Sometimes, I need to find out where the error is when mapping the JSON to a struct, which helps me out.